File: src/compiler/interpreter/OperatorUsageUndefined.java
package compiler.interpreter;

import compiler.lexer.token.OperatorToken;
import compiler.lexer.token.TypeToken;
import compiler.parser.UserException;

public class OperatorUsageUndefined extends UserException {
  public OperatorUsageUndefined(OperatorToken operator, TypeToken left, TypeToken right) {
    super("Operator " + operator.getValue() + " isn't defined between " + left.getValue() + " and " + right.getValue(), left);
  }
}

File: src/compiler/interpreter/OperatorUsageUndefined.kt
package compiler.interpreter

import compiler.parser.Symbols.Terminal.Operator
import compiler.parser.UserException

class OperatorUsageUndefined(operator: Operator, lvalue: Any, rvalue: Any) :
        UserException("Operator ${operator.str} isn't defined between ${lvalue.javaClass.simpleName} and ${rvalue.javaClass.simpleName}", operator)
